### PR TITLE
Slider branch

### PR DESCRIPTION
**Pull request by maywanting**
- kill some variable and create a class to display the data
- add the radio (setting the other data file of  convection and convection diffusion will be OK, perhaps.)

### CODE REVIEW

File: front/src/main.js [hunk 1]
```diff
@@ -48,131 +48,170 @@ const layout = {
...
 
-const dataName = 'data.json';
-// const dataName = 'convection-diffusion.json';
-const scalarFetch = fetch(dataName).then(response => response.json());
-const gradientFetch = fetch('gradient-fields.json').then(response => response.json());
-let intervalId = 0;
+let renderer = {
+    type : "cycle", //0-cycle, 1-slider
+    dataType : 0, //0-diffusion 1-convaction 2-convaction-diffusion
+    dataName : [ "diffusion", "convaction", "convaction-diffusion"],
```
Comment (kozenumezawa) [inline front/src/main.js]: typo

File: front/src/main.js [hunk 2]
```diff
@@ -48,131 +48,170 @@ const layout = {
...
+    typeFile : [
+        ["data.json", "gradient-fields.json"],
```
Comment (kozenumezawa) [inline front/src/main.js]: data.json is not a good file name, so please change "data.json" to "diffusion.json" by updating 'Python code'

File: front/public/index.html [hunk 3]
```diff
@@ -28,9 +28,17 @@
...
-            <div id="slider-value">0</div>
+        <div class="menu">
+            <div id="slider" height="500" width="500" display="inline-block">
+                <input type="range" id="slider-icon" min="0" max="50" step="1" value="0">
+                <span id="slider-value">0</span>
+            </div>
+            <button id="auto">auto</button>
+            <div id="typeChange">
+                <label> <input type="radio" name="datatype" value="0" checked="true">diffusion </label>
+                <label> <input type="radio" name="datatype" value="1">convaction</label>
```
Comment (kozenumezawa) [inline front/public/index.html]: Please fix this typo: convaction -> convection

File: front/src/main.js [hunk 4]
```diff
@@ -48,131 +48,170 @@ const layout = {
...
 
-
 Plotly.newPlot('plotlyDiv', [], layout);
 
-const dataName = 'data.json';
-// const dataName = 'convection-diffusion.json';
-const scalarFetch = fetch(dataName).then(response => response.json());
-const gradientFetch = fetch('gradient-fields.json').then(response => response.json());
-let intervalId = 0;
+let renderer = {
```
Comment (kozenumezawa) [inline front/src/main.js]: let -> const ?

File: front/src/main.js [hunk 5]
```diff
@@ -48,131 +48,170 @@ const layout = {
...
-    ctx.beginPath();
-
-    const SCALE = 5;
-    for (let x = 0; x < 100; x++) {
-        for (let y = 0; y < 100; y++) {;
-            ctx.fillStyle = colors[parseInt(scalarField[x][y])];
-            ctx.fillRect(x * SCALE, y * SCALE, SCALE, SCALE);
+        const SCALE = 5;
+        for (let x = 0; x < 100; x++) {
+            for (let y = 0; y < 100; y++) {;
```
Comment (kozenumezawa) [inline front/src/main.js]: meaningless ';' at the end of this line

File: front/src/main.js [hunk 6]
```diff
@@ -48,131 +48,170 @@ const layout = {
...
+    renderer.printGraph(index);
+});
+
+$("#auto").on("click", function() {
+    renderer.cycleInit();
+});
+
+$("#typeChange input").on("change", async function() {
+    let datatype = $("input[name='datatype']:checked").val();
+    if (datatype !== renderer.datatype && renderer.type == "cycle") {
```
Comment (kozenumezawa) [inline front/src/main.js]: Please change '==' to '==='

**Review by kozenumezawa — COMMENTED** [review]
Good work!

**Comment by kozenumezawa** [comment]
There are critical two bugs:
1. When 'auto button' is clicked multiple times, animation does not work well
2. Displayed data is not switched correctly when radio button is changed from 'convection-diffusion' to 'diffusion'
Please fix these :)

File: front/public/index.html (hunk 3 above)
Comment (maywanting) [reply front/public/index.html]: 厳しいな。。。

File: front/src/main.js (hunk 5 above)
Comment (maywanting) [reply front/src/main.js]: ";" has been removed

File: front/src/main.js (hunk 6 above)
Comment (maywanting) [reply front/src/main.js]: finished

File: front/src/main.js (hunk 2 above)
Comment (maywanting) [reply front/src/main.js]: finished

**Comment by maywanting** [comment]
- fix the bug of "auto button"
- fix the bug that convection diffusion can not be displayed

**Comment by kozenumezawa** [comment]
Data isn't switched correctly when the animation is stopped 
bug

**Comment by maywanting** [comment]
I also found this bug. last push has fixed this.

**Comment by kozenumezawa** [comment]
Looks good to me!!

**Pull request merged**